project-koku/koku
# [COST-1334] Sorting with limit/offset/order_by fix

This change addresses an issue where sorting with limit/offset/order_by would only sort each individual page by the order by parameter, sorting the whole list by default sorting mechanisms. This set of parameters now produces a result that orders the entire dataset by the order_by parameter. AWS is special and since account alias is a foreign key, it was sorting by the foreign key id instead of the actual account alias, that has also been addressed.

Testing: 
Since this change is a sorting change, the best way to test it out is to load up some data and try some sorts!

1. Load in some test data: `make create-test-customer` followed by `make load-test-customer-data`
2. Try an aws filter/offset/limit and check multiple pages and make sure everything is going in account alias order, such as: 

        - http://localhost:8000/api/cost-management/v1/reports/aws/costs/?filter[limit]=3&filter[offset]=0&group_by[account]=*&order_by[account_alias]=asc
        - http://localhost:8000/api/cost-management/v1/reports/aws/costs/?filter[limit]=3&filter[offset]=2&group_by[account]=*&order_by[account_alias]=asc
        - http://localhost:8000/api/cost-management/v1/reports/aws/costs/?filter[limit]=3&filter[offset]=4&group_by[account]=*&order_by[account_alias]=asc


3. Try the same with some of the other providers/order_by's, AWS account alias was the only oddball and had some extra changes related to this. Examples:
- `http://localhost:8000/api/cost-management/v1/reports/aws/costs/?filter[limit]=2&filter[offset]=0&group_by[account]=*&order_by[cost]=asc`
- `http://localhost:8000/api/cost-management/v1/reports/azure/costs/?filter[limit]=2&filter[offset]=0&group_by[subscription_guid]=*&order_by[subscription_guid]=asc`
- `http://localhost:8000/api/cost-management/v1/reports/openshift/costs/?filter[limit]=2&filter[offset]=0&group_by[project]=*&order_by[project]=asc`
- `http://localhost:8000/api/cost-management/v1/reports/gcp/costs/?filter[limit]=2&filter[offset]=0&group_by[project]=*&order_by[project]=asc`
